python/pyspark/util.py

```diff
@@ -369,39 +367,49 @@ class InheritableThread(threading.Thread):
     This API is experimental.
     """
 
-    def __init__(self, target, *args, **kwargs):
+    def __init__(self, target: Callable, *args: Any, **kwargs: Any):
```

Comment:
@HyukjinKwon This signature might not be good because this is different from the signature of `Thread.__init__`.

```py
    def __init__(
        self,
        group: None = ...,
        target: Optional[Callable[..., Any]] = ...,
        name: Optional[str] = ...,
        args: Iterable[Any] = ...,
        kwargs: Optional[Mapping[str, Any]] = ...,
        *,
        daemon: Optional[bool] = ...,
    ) -> None: ...
```

I guess users can't specify `group`?

Reply:
Yeah. I will give a shot to deprecate this API by fixing it in Py4J in Apache Spark 3.3.